contracts/Core/CollectionManager.sol

```diff
@@ -372,6 +372,9 @@ contract CollectionManager is Initializable, CollectionStorage, StateManager, Co
         }
     }
 
+    /**
+     * @dev updates the collectionIdToLeafIdRegistryOfLastEpoch everytime a block is being confirmed
```

Comment:
not everytime but whenever a collection status is changed/new collection is created